onnxruntime/core/providers/cuda/math/softmax.cc

```diff
@@ -84,7 +100,12 @@ Status Softmax<T>::ComputeInternal(OpKernelContext* ctx) const {
   if (input_shape.Size() == 0)
     return Status::OK();
 
-  return SoftMaxComputeHelper<T, false>(X_data, input_shape, Y_data, CudnnHandle(), axis_);
+  if (log_softmax_) {
```

Comment:
>log_softmax_ [](start = 6, length = 12)

when you have log_softmax_ as template parameter, you can directly call SoftMaxComputeHelper<T, is_log_softmax>